### PR TITLE
Add Support for SetTransformer++, ScheduleFree, Complexity Control. Improve stability, 2D Analysis, Small Fixes, RL Specialization Prototype

### PR DESCRIPTION
**Pull request by psaegert**
## TODO

- [x] Test base evaluation (run evaluation, load and verify results)
- [x] Add tests
  - [x] Evaluation with complexity conditioning
  - [x] Softmax sampling workflow
  - [x] Other missing coverage
  - [x] Dataset collate function for single instance "batches"
- [x] Explore edgecases
- [x] Finalize for merge
  - [x] Check if README is up-to-date
  - [x] Bump version

### CODE REVIEW

File: experimental/troubleshoot/float2bit_nan_shape.ipynb [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    "        fin_m = m2[..., :num_m_bits]\n",
+    "\n",
+    "        normal_result = torch.cat([s, e, fin_m], dim=-1).type(dtype)\n",
+    "        result[is_normal] = normal_result\n",
+    "\n",
+    "    # Handle NaN\n",
+    "    if torch.any(is_nan):\n",
+    "        # Set all exponent bits to 1 and non-zero mantissa (conventionally first mantissa bit is 1)\n",
+    "        # nan_pattern = torch.zeros(result.shape[:-1]).type(dtype)\n",
+    "        nan_pattern = torch.zeros(num_e_bits + num_m_bits + 1).type(dtype)\n",
```
[the remark this answers is not in the document]
Comment (psaegert) [reply experimental/troubleshoot/float2bit_nan_shape.ipynb]: Resolved in [427689b](https://github.com/psaegert/flash-ansr/pull/19/commits/427689ba0223ca74ae7f90d2862cc42682a8b05b)

File: src/flash_ansr/models/manage.py [hunk 2]
```diff
@@ -14,19 +14,19 @@ def install_model(model: str, local_dir: str | None = None, verbose: bool = True
         print(f"Model {model} installed successfully!")
 
 
-def remove_model(path: str, verbose: bool = True, force_remove: bool = False) -> None:
+def remove_model(path: str, verbose: bool = True) -> None:
```
[the remark this answers is not in the document]
Comment (psaegert) [reply src/flash_ansr/models/manage.py]: Resolved in [427689b](https://github.com/psaegert/flash-ansr/pull/19/commits/427689ba0223ca74ae7f90d2862cc42682a8b05b)
